flink-tests/src/test/java/org/apache/flink/runtime/metrics/JobManagerMetricsITCase.java

```diff
@@ -0,0 +1,180 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.metrics;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.testutils.BlockerSync;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.MetricConfig;
+import org.apache.flink.metrics.reporter.AbstractReporter;
+import org.apache.flink.runtime.client.JobExecutionException;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.PrintSinkFunction;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+import org.apache.flink.test.util.MiniClusterResource;
+import org.apache.flink.test.util.MiniClusterResourceConfiguration;
+
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Integration tests for proper initialization of the job manager metrics.
+ */
+public class JobManagerMetricsITCase {
+
+	private static final String JOB_MANAGER_METRICS_PREFIX = "localhost.jobmanager.";
+
+	private static final BlockerSync sync = new BlockerSync();
+
+	@ClassRule
+	public static final MiniClusterResource MINI_CLUSTER_RESOURCE = new MiniClusterResource(
+		new MiniClusterResourceConfiguration.Builder()
+			.setConfiguration(getConfiguration())
+			.setNumberTaskManagers(1)
+			.setNumberSlotsPerTaskManager(1)
+			.build());
+
+	@Before
+	public void setUp() throws Exception {
+		Thread jobExecuteThread = new Thread(() -> {
+
+			StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+			env.addSource(new SourceFunction<String>() {
+
+				@Override
+				public void run(SourceContext<String> ctx) throws Exception {
+					sync.block();
+				}
+
+				@Override
+				public void cancel() {
+					sync.releaseBlocker();
+				}
+
+			}).addSink(new PrintSinkFunction());
+
+			try {
+				env.execute();
+			} catch (Exception e) {
+				if (!(e instanceof JobExecutionException)) {
+					fail("Execution job failed, reason : " + e.getMessage());
```

Comment:
This won't have the effect you probably intended, as it will not fail the test. You can use a Flink `CheckedThread` to achieve this, and `sync()` at the end of the test.